api/src/components/image/controller.ts

```diff
@@ -7,18 +7,24 @@ export const create = async (event: I.AuthenticatedAPIRequest<I.ImageSentBody>):
     try {
         const imageType = event.body.image.split(';')[0].split('/')[1];
 
-        if (imageType !== 'png' && imageType !== 'jpeg' && imageType !== 'jpg') {
-            return response.json(422, { message: 'Invalid' });
+        if (!I.ImageExtension[imageType]) {
+            return response.json(422, {
+                message: `Failed to upload "${event.body.name}". The format is not supported.`
+            });
         }
 
-        const imageReference = await imageService.createDBReference(event.body.name, imageType, event.user.id);
+        const imageReference = await imageService.createDBReference(
+            event.body.name,
+            imageType as I.ImageExtension,
```

Comment:
Yes, we need the cast because TS is not smart enough in this specific case
<img width="1176" alt="image" src="https://user-images.githubusercontent.com/48569671/214541603-a7d6836c-375c-4868-8715-2d713076eacc.png">